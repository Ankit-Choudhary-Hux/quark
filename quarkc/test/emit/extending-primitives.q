namespace pkg {

    @mapping($java{java.lang.Runnable})
    primitive Runnable {
        void event1() {
            print("default event1");
        }
        void run() {
            print("running");
        }
    }

    @mapping($java{java.lang.Cloneable})
    primitive Cloneable {
        void event2() {
            print("default event2");
        }
    }

    class C extends Runnable, Cloneable {
        void event1() {
            print("event1");
        }
    }

}

void main() {
    pkg.C c = new pkg.C();
    c.event1();
    c.event2();
    c.run();
}
