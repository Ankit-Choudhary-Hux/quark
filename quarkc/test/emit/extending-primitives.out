event1
default event2
running
